tests/contrib/funsor/test_vectorized_markov.py

```diff
@@ -457,3 +457,107 @@ def test_vectorized_markov_multi(model, weeks_data, days_data, vars1, vars2, his
 
         assert actual_weeks_step == expected_weeks_step
         assert actual_days_step == expected_days_step
+
+
+def guide_empty(data, history, vectorized):
+    pass
+
+
+@pytest.mark.parametrize("model,guide,data,history", [
+    (model_0, guide_empty, torch.rand(3, 5, 4), 1),
+    (model_1, guide_empty, torch.rand(5, 4), 1),
+    (model_2, guide_empty, torch.ones((5, 4), dtype=torch.long), 1),
+    (model_3, guide_empty, torch.ones((5, 4), dtype=torch.long), 1),
+    (model_4, guide_empty, torch.ones((5, 4), dtype=torch.long), 1),
+    (model_5, guide_empty, torch.ones((5, 4), dtype=torch.long), 2),
+    (model_6, guide_empty, torch.rand(5, 4), 1),
+    (model_6, guide_empty, torch.rand(100, 4), 1),
+    (model_7, guide_empty, torch.ones((5, 4), dtype=torch.long), 1),
+    (model_7, guide_empty, torch.ones((50, 4), dtype=torch.long), 1),
+])
+def test_model_enumerated_elbo(model, guide, data, history):
+
+    with pyro_backend("contrib.funsor"):
+        if history > 1:
+            pytest.xfail(reason="TraceMarkovEnum_ELBO does not yet support history > 1")
+
+        elbo = infer.TraceEnum_ELBO(max_plate_nesting=4)
+        expected_loss = elbo.differentiable_loss(model, guide, data, history, False)
+
+        vectorized_elbo = infer.TraceMarkovEnum_ELBO(max_plate_nesting=4)
+        actual_loss = vectorized_elbo.differentiable_loss(model, guide, data, history, True)
+
+        assert_close(actual_loss, expected_loss)
```

Comment:
These tests look good, but can you also check gradients?
```py
expected_loss = ...
params = pyro.get_param_store().values()
expected_grads = torch.autograd.grad(expected_loss, params)

vectorized_elbo = ...
actual_loss = ...
actual_grads = torch.autograd.grad(actual_loss, params)

assert_close(actual_loss, expected_loss)
for actual_grad, expected_grad in zip(actual_grads, expected_grads):
    assert_close(actual_grad, expected_grad)
```

Reply:
Using `torch.autograd.grad` gives `RuntimeError`:

```
>       return Variable._execution_engine.run_backward(
            outputs, grad_outputs_, retain_graph, create_graph,
            inputs, allow_unused)
E       RuntimeError: One of the differentiated Tensors appears to not have been used in the graph. Set allow_unused=True if this is the desired behavior.
```

Do I have to add `clear_param_store()` in the models?

Reply:
You could add `clear_param_store()` to the beginning of the test body.